_toolboxes/image-steganography.md

```diff
@@ -0,0 +1,89 @@
+
+Additionally, since changing the LSB doesn't drastically change the overall value of the of 8 bit number, we can hide our data without modifying a source image in any detectable sort of way. You can test this out with any [RGB color wheel](http://www.colorspire.com/rgb-color-wheel/) to get a sense of how little difference there is between a color like (150, 50, 50) and (151, 50, 50)
+
+
+**Aside**
+
+The concept of MSB and LSB occurs in other contexts as well. For example, [parity bits](https://en.wikipedia.org/wiki/Parity_bit) are used as a basic form of error checking. Additionally, because the LSBs will change rapidly even if the value of the bit changes a little, they are very useful for use in [hash functions](https://en.wikipedia.org/wiki/Hash_function) and [checksums](https://en.wikipedia.org/wiki/Checksum) for validation purposes.
+
+## Decoding the sample image
+
+Provided in this toolbox is a picture of a cute dog. However, this dog is hiding a very secret message... can you decode it?  
+
+![](../images/toolboxes/image-steganography/encoded_sample.png)
+
+Provided in the starter code is a function called `decode_image()`. The secret image was hidden in the LSB of the pixels in the red channel of the image. That is, the value of the LSB of each red pixel is 1 if the hidden image was 1 at that location, and 0 if the hidden image was also 0. Your task is to iterate though each pixel in the encoded image and set the decode_image pixel to be (0, 0, 0) or (255, 255, 255) depending on the value of that LSB.
+
+You may want to look at the Python [bin](https://docs.python.org/3/library/functions.html#bin) function as you convert between integer and binary. Remember that bin will convert an integer to a *binary string*. Also, remember that you have to isolate the red_channel from the original RGB image. You can do this using the .split() function that PIL provides.
+
+```python
+def decode_image(file_location):
+    encoded_image = Image.open(file_location)
+    red_channel = encoded_image.split()[0]
+
+    x_size = encoded_image.size[0]
+    y_size = encoded_image.size[1]
+
+    decoded_image = Image.new("RGB", encoded_image.size)
+    pixels = decoded_image.load()
+
+    #TODO: Fill in decoding functionality
+
+    decoded_image.save("images/decoded_image.png")
+```
+
+## Encoding a secret message
+
+Now that we can decode secret messages, its only natural that we want to encode some too! Provided in the starter code are a pair of functions called `write_text()` and `encode_image()`. `write_text()` will take a string and convert it to a black and white image of the string. You may use it as a helper function in completing your implementation of `encode_image()`.
+
+## Completing the Toolbox Exercise
+
+To turn in your assignment, push your code, decoded image the cute puppy is hiding, and a sample of an encoded secret message to GitHub and submit a pull request.
```

Comment:
How about:

You will need three things to complete this assignment:

1. Completed steganography.py
2. Decoded image the cute puppy is hiding
3. A sample of an encoded image with your hidden message

Commit all three things to your repo and submit a pull request to turn in this toolbox.